### PR TITLE
DNS Record: resource and data source

### PR DESCRIPTION
**Pull request by mathewab**

### CODE REVIEW

File: examples/data-sources/bloxone_dns_a_records/data-source.tf [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Get "A" records filtered by an attribute
+data "bloxone_dns_a_records" "example_by_attribute" {
+  filters = {
+    "absolute_name_spec" = "abc.example.com"
```
Comment (venkat-iblox) [inline examples/data-sources/bloxone_dns_a_records/data-source.tf:4]: Do we need the quotes around the 'key'?

File: examples/data-sources/bloxone_dns_records/data-source.tf [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# Get records filtered by an attribute
+data "bloxone_dns_records" "example_by_attribute" {
```
Comment (venkat-iblox) [inline examples/data-sources/bloxone_dns_records/data-source.tf:2]: would this handle the `generic` record in addition to everything?

File: internal/service/dns_data/api_record_resource.go [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+	r.client = client
+}
+
+func (r *RecordResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_" + r.impl.resourceName()
+}
+
+func (r *RecordResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "",
```
Comment (venkat-iblox) [inline internal/service/dns_data/api_record_resource.go:66]: No Description?

Comment (mathewab) [reply internal/service/dns_data/api_record_resource.go]: Added

File: examples/data-sources/bloxone_dns_records/data-source.tf (hunk 2 above)
Comment (mathewab) [reply examples/data-sources/bloxone_dns_records/data-source.tf:2]: No, it is supposed to handle only types not supported directly. I have added a `type` field in the data source for this.

File: examples/data-sources/bloxone_dns_a_records/data-source.tf (hunk 1 above)
Comment (mathewab) [reply examples/data-sources/bloxone_dns_a_records/data-source.tf:4]: It is not needed, but even if it is present, it is not a problem.

**Review by venkat-iblox — APPROVED** [review]

**Pull request merged**